active/tab/tabcontrol.md

```diff
@@ -0,0 +1,401 @@
+        <KeyboardAccelerator Key="Number9" Modifiers="Control" Invoked="NavigateToNumberedTabKeyboardAccelerator_Invoked" />
+    </controls:TabView.KeyboardAccelerators>
+    <!-- ... some tabs ... -->
+</controls:TabView>
+```
+
+``` csharp
+
+private void NewTabKeyboardAccelerator_Invoked(KeyboardAccelerator sender, KeyboardAcceleratorInvokedEventArgs args)
+{
+    // See previous sample
+    CreateNewTab();
+}
+
+private void CloseSelectedTabKeyboardAccelerator_Invoked(KeyboardAccelerator sender, KeyboardAcceleratorInvokedEventArgs args)
+{
+    // Only close the selected tab if it is closeable
+    if (((TabViewItem)TabRoot.SelectedItem).IsCloseable)
+    {
+        TabRoot.Items.Remove(TabRoot.SelectedItem);
+    }
+}
+
+private void NavigateToNumberedTabKeyboardAccelerator_Invoked(KeyboardAccelerator sender, KeyboardAcceleratorInvokedEventArgs args)
+{
+    int tabToSelect = 0;
+
+    switch (sender.Key)
+    {
+        case Windows.System.VirtualKey.Number1:
+            tabToSelect = 0;
+            break;
+        case Windows.System.VirtualKey.Number2:
+            tabToSelect = 1;
+            break;
+        case Windows.System.VirtualKey.Number3:
+            tabToSelect = 2;
+            break;
+        case Windows.System.VirtualKey.Number4:
+            tabToSelect = 3;
+            break;
+        case Windows.System.VirtualKey.Number5:
+            tabToSelect = 4;
+            break;
+        case Windows.System.VirtualKey.Number6:
+            tabToSelect = 5;
+            break;
+        case Windows.System.VirtualKey.Number7:
+            tabToSelect = 6;
+            break;
+        case Windows.System.VirtualKey.Number8:
+            tabToSelect = 7;
+            break;
+        case Windows.System.VirtualKey.Number9:
+            // Select the last tab
+            tabToSelect = TabRoot.Items.Count - 1;
+            break;
+    }
+
+    // Only select the tab if it is in the list
+    if (tabToSelect < TabRoot.Items.Count)
+    {
+        TabRoot.SelectedIndex = tabToSelect;
+    }
+}
+
+```
+
+## Combine tabs from one window into another window
+TODO -- should this be part of the "tab tear out" sample above?
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples
+section.  For example, see the Remarks for the MediaPlayerElement 
+(https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.MediaPlayerElement#remarks). -->
+
+## Keyboarding behavior
+### TAB and Arrow key behavior
+The TabView has three content areas: TabStrip, TabStripHeader (to the left of the TabStrip) and TabStripFooter (to the right of the TabStrip). When the user hits the TAB key, focus moves between the content areas. When focus moves into the Tab content area, the selected Tab gains focus. Once the user is in the Tab content area, pressing the arrow key will move focus between the tabs. Arrow focus is trapped inside the Tab content area.
```

Comment:
I'm having trouble parsing this. It doesn't necessarily need to be longer, or if it is, maybe move it to the appendix. Right now I'm just not sure what it's saying.

"When the user hits the TAB key, focus moves between the content areas." ... Doesn't it move _within_ the content area too?

"When focus moves into the Tab content area, the selected Tab gains focus" ... How do you move into the tab content area without selecting the tab first.  And does "tab content area" mean the Content of a TabViewItem? In Edge you use arrows to tab between the tab item headings, then Enter/Space to select a tab.

Reply:
> "When the user hits the TAB key, focus moves between the content areas." ... Doesn't it move within the content area too?

In the left and right content areas which have app-provided content, TAB will naively do whatever the app says (which will be moving between the elements in those areas by default). However, the behavior is different in the tab strip area. When focus is outside of the tab strip area and the user presses TAB, it moves focus to the selected tab. If the user presses TAB again, it will move focus out of the tab strip area. If focus is within the tab strip area and the user presses LEFT/RIGHT arrows, it will move focus (not selection). Once focus has moved, the user can press ENTER/SPACE to select the focused item. 

I shouldn't have said "Tab content area" - I meant "tab strip". I'll clarify. 

The TL;DR is: we're trying to mimic what Edge does.